packages/server/lib/project-base.ts

```diff
@@ -512,21 +516,23 @@ export class ProjectBase<TServer extends ServerE2E | ServerCt> extends EE {
 
     const obj = {
       onChange: () => {
+        debug('settings changed')
         // dont fire change events if we generated
         // a project id less than 1 second ago
         if (this.generatedProjectIdTimestamp &&
           ((Date.now() - this.generatedProjectIdTimestamp) < 1000)) {
           return
         }
 
+        debug('updating the project re-running')
         // call our callback function
         // when settings change!
         onSettingsChanged()
       },
     }
 
     if (configFile !== false) {
-      this.watchers.watch(settings.pathToConfigFile(projectRoot, { configFile }), obj)
+      this.watchers.watchTree(settings.pathToConfigFile(projectRoot, { configFile }), obj)
```

Comment:
This looks suspicious. Why did this line change?

Reply:
With this PR we can have evaluated JavaScript resulting in a config object. Watching the settings could become watching more than one file when the targeted file imports/requires dependencies.

This allows us to watch the file given and its dependencies.

Reply:
I think it's in order to watch not only `cypress.config.js` but any files it imports as well.

Reply:
@lmiller1990 your english is light years ahead of mine.

Reply:
Je ne parle pas français